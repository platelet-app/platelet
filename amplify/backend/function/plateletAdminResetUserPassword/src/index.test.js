const indexModule = require("./index");
const aws = require("aws-sdk");

jest.mock(
    "/opt/appSyncRequest",
    () => {
        return {
            request: jest.fn(),
            errorCheck: jest.fn(),
        };
    },
    { virtual: true }
);
jest.mock(
    "/opt/graphql/queries",
    () => {
        return {
            getUser: "getUser",
        };
    },
    { virtual: true }
);

const appsyncModule = require("/opt/appSyncRequest");

jest.mock(
    "/opt/sendWelcomeEmail",
    () => {
        return {
            sendWelcomeEmail: jest.fn(),
        };
    },
    { virtual: true }
);

const sendWelcomeModule = require("/opt/sendWelcomeEmail");
// mock CognitoIdentityServiceProvider class
//
const cognitoIdentityServiceProvider = Object.getPrototypeOf(
    new aws.CognitoIdentityServiceProvider()
);

describe("plateletAdminResetUserPassword", () => {
    const originalRandom = global.Math.random;
    afterEach(() => {
        global.Math.random = originalRandom;
    });
    beforeEach(() => {
        global.Math.random = () => 0.5;
    });
    test("reset the user's password", async () => {
        const event = {
            arguments: {
                userId: "user-id",
            },
        };
        appsyncModule.request.mockReturnValueOnce({
            json: () => ({
                data: {
                    getUser: {
                        name: "Some name",
                        contact: {
                            emailAddress: "some@example.com",
                        },
                        username: "someusername",
                    },
                },
            }),
        });
        jest.spyOn(
            cognitoIdentityServiceProvider,
            "adminSetUserPassword"
        ).mockReturnValue({
            promise: () => Promise.resolve(),
        });
        await indexModule.handler(event);
        expect(appsyncModule.request).toMatchSnapshot();
        expect(sendWelcomeModule.sendWelcomeEmail).toMatchSnapshot();
    });
});
