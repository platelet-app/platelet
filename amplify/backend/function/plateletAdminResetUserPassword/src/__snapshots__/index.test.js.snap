// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plateletAdminResetUserPassword reset the user's password 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "query": "getUser",
        "variables": Object {
          "id": "user-id",
        },
      },
      "https://api.example.com/graphql",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Object {
        "json": [Function],
      },
    },
  ],
}
`;

exports[`plateletAdminResetUserPassword reset the user's password 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "some@example.com",
      "Some name",
      "i",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
